libtransact/src/state/merkle.rs

```diff
@@ -140,18 +148,24 @@ impl Prune for MerkleDatabase {
     }
 }
 
+/// Merkle Database
+#[derive(Clone)]
+pub struct MerkleDatabase {
+    root_hash: String,
+    root_node: Node,
+}
+
 impl MerkleDatabase {
     /// Constructs a new MerkleDatabase, backed by a given Database
     ///
     /// An optional starting merkle root may be provided.
-    pub fn new(db: LmdbDatabase, merkle_root: Option<&str>) -> Result<Self, StateDatabaseError> {
+    pub fn new(db: &LmdbDatabase, merkle_root: Option<&str>) -> Result<Self, StateDatabaseError> {
         let root_hash = merkle_root.map_or_else(|| initialize_db(&db), |s| Ok(s.into()))?;
         let root_node = get_node_by_hash(&db, &root_hash)?;
 
         Ok(MerkleDatabase {
-            root_hash: Arc::new(Mutex::new(root_hash)),
-            root_node: Arc::new(Mutex::new(root_node)),
-            db,
```

Comment:
MerkleDB should still own a copy of the DB - it is a thread-safe, cloneable structure.  Think of this api as one that will be used outside of the transact internals, where as implementing the Read, Write, Prune traits are for transact-specific operations